lib/oauth2_client.dart

```diff
@@ -41,14 +47,18 @@ class OAuth2Client {
   Map<String, String> _accessTokenRequestHeaders;
 
   WebAuth webAuthClient;
+  CredentialsLocation credentialsLocation =
+      CredentialsLocation.CREDENTIALS_LOCATION_HEADER;
 
   OAuth2Client(
       {@required this.authorizeUrl,
       @required this.tokenUrl,
       this.refreshUrl,
       this.revokeUrl,
       @required this.redirectUri,
-      @required this.customUriScheme}) {
+      @required this.customUriScheme,
+      this.credentialsLocation,
```

Comment:
... otherwise, if the ``credentialsLocation`` param isn't passed to the constructor, the class property will be overridden to null.
```suggestion
      this.credentialsLocation = CredentialsLocation.HEADER,
```